### PR TITLE
[opentelemetry-cpp] update to 1.12.0

### LINKED ISSUE #33772: [opentelemetry-cpp] update to 1.11.0
### Library name

opentelemetry-cpp

### New version number

1.11.0

### Other information that may be useful (release notes, etc...)

vcpkg has 1.9.1, which is 2 releases behind.

https://github.com/open-telemetry/opentelemetry-cpp/releases/tag/v1.11.0

### PR DESCRIPTION
**Pull request by jimwang118**
Fixes https://github.com/microsoft/vcpkg/issues/33772

1. Update to 1.12.0.
2. Delete feature `otlp` based on upstream PR [2161](https://github.com/open-telemetry/opentelemetry-cpp/pull/2161).
3. `opentelemetry-fluentd` adds missing header files.
4. Delete the upstream fixed patch `use-default-cxx-version.patch and add-missing-include-file.patch`.

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

Usage test pass with following triplets:

```
x86-windows
x64-windows
x64-windows-static
```

### CODE REVIEW

**Comment by PriitUring** [comment]
Title has typo: `opentelemtery-cpp`, correct would be `opentelemetry-cpp`?

File: ports/opentelemetry-cpp/portfile.cmake [hunk 1]
```diff
@@ -28,7 +27,6 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
         prometheus WITH_PROMETHEUS
         elasticsearch WITH_ELASTICSEARCH
         jaeger WITH_JAEGER
-        otlp WITH_OTLP
```
Comment (BillyONeal) [inline ports/opentelemetry-cpp/portfile.cmake:31]: What happened to `WITH_OLTP`?

If it's gone the feature should go too.

**Review by BillyONeal — DISMISSED** [review]

File: ports/opentelemetry-cpp/portfile.cmake (hunk 1 above)
Comment (jimwang118) [reply ports/opentelemetry-cpp/portfile.cmake:31]: Delete feature otlp based on upstream PR [2161](https://github.com/open-telemetry/opentelemetry-cpp/pull/2161.)

**Review by Adela0814 — DISMISSED** [review]

File: ports/opentelemetry-cpp/portfile.cmake (hunk 1 above)
Comment (BillyONeal) [reply ports/opentelemetry-cpp/portfile.cmake:31]: I see. Based on that PR does it seem reasonable that `vcpkg.json` is changed to:

```json
    "otlp": {
      "description": "Whether to include the OpenTelemetry Protocol in the SDK",
      "dependencies": [
        {"name": "opentelemetry-cpp", "default-features": false, "features": ["otlp-grpc", "otlp-http"]}
      ]
    }
```

If not I think the oltp feature needs to be removed from vcpkg.json

**Comment by dbolduc** [comment]
Ping. I selfishly want to use some of the features in opentelemetry-cpp v1.12.0 :)

**Comment by jimwang118** [comment]
> Ping. I selfishly want to use some of the features in opentelemetry-cpp v1.12.0 :)

OK, check that the latest version upstream is v1.12.0, I will update to the latest version.

File: versions/o-/opentelemetry-cpp.json [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "5fd8cacd9fd3aaa30b0ad133f2de052ba3ff869f",
+      "version-semver": "1.12.0",
+      "port-version": 0
+    },
+    {
+      "git-tree": "73b513869768adc1fa546e6c96b823694ece9f08",
+      "version-semver": "1.11.0",
+      "port-version": 0
+    },
```
Comment (dbolduc) [inline versions/o-/opentelemetry-cpp.json:12]: I have no power here... but it looks like this `1.11.0` is an artifact that should not be included. Maybe revert the file and redo the `./vcpkg x-add-version --all --overwrite-version` ?

---

Thanks for updating to 1.12.0 by the way!

Comment (Adela0814) [inline versions/o-/opentelemetry-cpp.json:12]:
```suggestion
```

**Review by Adela0814 — CHANGES_REQUESTED** [review]

**Review by Adela0814 — APPROVED** [review]

**Review by data-queue — APPROVED** [review]

File: ports/opentelemetry-cpp/vcpkg.json [hunk 3]
```diff
@@ -39,7 +38,14 @@
     "otlp": {
       "description": "Whether to include the OpenTelemetry Protocol in the SDK",
       "dependencies": [
-        "protobuf"
```
Comment (ScottEllis1) [inline ports/opentelemetry-cpp/vcpkg.json:42]: @jimwang118 

What was the reason for removing this and adding the other features as dependencies? Now gRPC is included if otlp-http is included. I was previously trying to limit the size of the dependencies and I didn't want to include gRPC if I didn't have to. reverting these lines seems to fix my problem. Was this intended?

**Pull request merged**